src/test/java/com/acrolinx/proxy/util/HttpMethod.java

```diff
@@ -40,7 +40,7 @@ public void callAcrolinxProxyMethod(
         acrolinxProxy.doPut(httpServletRequest, httpServletResponse);
         break;
       default:
-        Assertions.fail("Unknown HTTP method: " + this);
+        throw new NotImplementedException("Unknown HTTP method: " + this);
```

Comment:
I would suggest that you use a custom exception. Suppose we remove the dependency `org.apache.commons.lang3` it would mean that we have to come here and change this again.

Also, remember to add the exception to the method signature.

Reply:
@willynganga Makes sense, however for such a small usecase, I think it is better to stick to exceptions that are already at our disposal. How does IllegalArgumentException sound like?

Reply:
That also works 👍🏾